src/CP/valueselection/learning/rewards/smartreward.jl

```diff
@@ -38,16 +46,34 @@ end
 """
     set_reward!(::DecisionPhase, lh::LearnedHeuristic{SmartReward, O}, model::CPModel)
 
-Change the current reward at the DecisionPhase. This is called right before making the next decision, so you know you have the very last state before the new decision
-and every computation like fixPoints and backtracking has been done.
+Change the current reward at the DecisionPhase. This is called right before making the next decision, so you know you have the very last state before the new decision and every computation like fixPoints and backtracking has been done.
+
+This computes the reward : ρ*( 1+ tour_upper_bound  - last_dist) where ρ is a constant, tour_upper_bound and upper bound of the tour and lastdist the distance between the previous node and the target node decided by the previous decision (the reward is attributed just before takng a new decision)
 """
 function set_reward!(::Type{DecisionPhase}, lh::LearnedHeuristic{SR, SmartReward, A}, model::CPModel) where {
     SR <: AbstractStateRepresentation,
     A <: ActionOutput
 }
-    #println("Decision, reward : ",model.statistics.lastPruning)
+    dist = model.adhocInfo[1]
+    n =  size(dist)[1]
+
+    tour_upper_bound = Base.maximum(dist) * n
+    max_dist = Float32(Base.maximum(dist))
+
+    if !isnothing(model.statistics.lastVar)
+        x = model.statistics.lastVar
+        s = x.id
+        current = parse(Int,split(x.id,'_')[2])
+        if isbound(model.variables["a_"*string(current)])
+            a_i = assignedValue(model.variables["a_"*string(current)])
+            v_i = assignedValue(model.variables["v_"*string(current)])
+            last_dist = lh.current_state.dist[v_i, a_i] * max_dist
```

Comment:
This corresponds to the distance between the previous node and the node that has just been selected by the heuristic one step before. (Recall that the reward is always given one step after, just before making a new decision).